src/test/java/com/amazonaws/services/dynamodbv2/LockItemPaginatedQueryIteratorTest.java

```diff
@@ -0,0 +1,50 @@
+package com.amazonaws.services.dynamodbv2;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import junit.framework.TestCase;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.QueryRequest;
+import software.amazon.awssdk.services.dynamodb.model.QueryResponse;
+
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LockItemPaginatedQueryIteratorTest extends TestCase {
```

Comment:
Let's add more tests here. For example, next when there is a next. Also, tests for hasNext method are missing

Reply:
Added some tests.